packages/h5wasm/src/h5wasm-api.ts

```diff
@@ -59,7 +64,9 @@ export class H5WasmApi extends ProviderApi {
 
     return Object.fromEntries(
       Object.entries(h5wEntity.attrs).map(([name, attr]) => {
-        return [name, (attr as H5WasmAttribute).value];
+        assertH5WasmAttribute(attr);
```

Comment:
So `h5wEntity.attrs` is now typed as `{[key: string]: {get(): Attribute, enumerable: true}}`.

As a result, `attr` type gets infered as `{get(): Attribute, enumerable: true}` for some reason while the actual runtime object `attr` has the signature of `H5WasmAttribute`. 

There I cannot access `dtype`, `shape` or `value` without asserting first the type. I tried to find a more elegant workaround but could not.

Some relevant issues that I came across that could help understand:
- https://github.com/microsoft/TypeScript/issues/28694
- https://github.com/microsoft/TypeScript/issues/38115
- https://github.com/microsoft/TypeScript/issues/42919

Reply:
So is the type cast `as H5WasmAttribute` no longer working because TS thinks the types are too different? If so, then maybe `as unknown as H5WasmAttribute` would work? It's not very elegant, but it has the advantage of clearly pointing out the hack (as opposed to an assertion, which has a valid purpose - i.e. typically to narrow down something from a broader type). I'm not opposed to the assertion if you prefer it to a type cast, though.